src/autopas/utils/SimilarityFunctions.h

```diff
@@ -65,6 +65,12 @@ std::pair<double, double> calculateHomogeneityAndMaxDensity(const Container &con
     for (size_t i = 0; i < particleLocation.size(); i++) {
       index[i] = particleLocation[i] / cellLength;
     }
+
+    // ignore particles that are outside the domain
```

Comment:
Something like this is needed, as otherwise it is possible to access memory outside the `particlesPerCell` array. Wich causes the sph-mpi tests to fail.

Im not sure if just ignoring such particles is the right answer as it could mess with the calculated density/homogenity. 

But such cases shouldn't even happen in the first place and maybe there is something wrong in the way the index is calculated, but at least this fixes the memory access issues.